src/dc_receive_imf.rs

```diff
@@ -954,26 +954,35 @@ async fn add_parts(
                     }
                 };
 
+                // If you change which information is skipped if the message is trashed,
+                // also change `MsgId::trash()` and `delete_expired_messages()`
+                // (to find the exact location, you can grep for `MsgId::trash()`)
+                let shown = !chat_id.is_trash();
```

Comment:
the "shown" wording reminds me a bit too much to "hidden" - which is a different thing.

maybe just keep the wording "trash"? so, `let trash = chat_id.is_trash();` and swap if/else below.